proxy_map.go

```diff
@@ -1160,6 +1160,17 @@ func (m *Map) LocalMapStats() LocalMapStats {
 	return LocalMapStats{}
 }
 
+func (m *Map) destroyLocally(ctx context.Context) {
+	m.logger.Trace(func() string {
+		return fmt.Sprintf("hazelcast.Map.destroyLocally: %s", m.name)
+	})
+	if m.hasNearCache {
```

Comment:
I think the current state where we access a proxy, checking whether it is a map, checking whether it has near cache is too specific. Can we make these an interface or a field in the proxy struct?

What I want to say is that it would be great to make these part of the general flow. Wdyt?

Reply:
We can consider that if we implement the support for Replicated Map. There are no other data structures that support the near cache.